server/pulp/server/async/tasks.py

```diff
@@ -631,14 +631,18 @@ def _handle_cProfile(self, task_id):
             self.pr.dump_stats("%s/%s" % (profile_directory, task_id))
 
 
-def cancel(task_id):
+def cancel(task_id, worker_cleanup=False):
```

Comment:
What about making this `worker_cleanup=True` and making the if statement instead be `if worker_cleanup`.

I think that would be clearer because currently if worker_cleanup is False, it actually does cleanup.

Reply:
I see your point about this being unclear but I disagree that the logic should be changed (the naming, perhaps. I think you parsed it in the opposite way from what I intended).  

This is the exception, not the rule - only once place in the codebase attempts to cancel tasks during worker cleanup, but lots of places in the codebase attempt to cancel tasks during normal operation.  If we were to flip this default, we would then need to specify ```worker_cleanup=False``` on every call to ```cancel()``` but this one, and remember to do so in the future.

Reply:
The changes I'm suggestion would keep it as the exception not the rule because in addition to changing the default value from False to True you would also change the s/if not worker_cleanup/if worker_cleanup/

Since the default would be to perform cleanup, we would only need to specify `worker_cleanup=False` on that one call that wants to modify the "cleanup by default" behavior.

Reply:
Ah, so this is basically switching the semantic meaning of ```worker_cleanup```.  That is reasonable, I'll change it.

Reply:
Exactly! Thank you.